content/shmem_team_collect.tex

```diff
@@ -0,0 +1,96 @@
+void @\FuncDecl{shmem\_team\_collect32}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
+void @\FuncDecl{shmem\_team\_collect64}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
+void @\FuncDecl{shmem\_team\_fcollect32}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
+void @\FuncDecl{shmem\_team\_fcollect64}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
+\end{Csynopsis}
+
+\begin{apiarguments}
+
+\apiargument{IN}{team}{A valid \openshmem team handle to a team which has been
+    created without disabling support for collective operations.}
+\apiargument{OUT}{dest}{A symmetric array large enough
+    to accept the concatenation of the \source{} arrays on all \acp{PE} in the team.
+    See table below in this description for allowable data types.}
+\apiargument{IN}{source}{A symmetric data object that can be of any type permissible
+    for the \dest{} argument.}
+\apiargument{IN}{nelems}{The number of elements in the \source{} array. \VAR{nelems}
+    must be of type \VAR{size\_t}.}
+
+\end{apiarguments}
+
+\apidescription{   
+    \openshmem \FUNC{team\_collect} and \FUNC{team\_fcollect} are collective routines
+    over an existing team. These routines concatenate \VAR{nelems}
+    \CONST{64}-bit or \CONST{32}-bit data items from the \source{} array into the
+    \dest{} array, over all \acp{PE} in the specified \VAR{team} in processor number order.
+    The resultant \dest{} array contains the contribution from the first \ac{PE} in the \VAR{team}
+    , then the contribution from the second \ac{PE} in the \VAR{team}, and so on.
+    The collected result is written to the \dest{} array for all \acp{PE} in the team.
+    
+    The \FUNC{fcollect} routines require that \VAR{nelems} be the same value in all
+    \acp{PE} in the team, while the \FUNC{collect} routines allow \VAR{nelems} to
+    vary from \ac{PE} to \ac{PE}.
+
+    As with all \openshmem team collective routines, each of these routines assumes that
+    only \acp{PE} in the given team call the routine.  If a \ac{PE} not in the
+    team calls an \openshmem team collective routine, the behavior is undefined.
+
+    If the team has been created with the \LibConstRef{SHMEM\_TEAM\_NOCOLLECTIVE} option,
+    it will not have the required support structures to complete this routine. If
+    such a team is passed to this or any other team collective routine, the behavior
+    is undefined.
+
+    The same \dest{} and \source{} data objects must be passed by all \acp{PE}
+    in the team.
+
+    Upon return from a collective routine, the following are true for the local
+    \ac{PE}:
+    \begin{itemize}
+    \item The \dest{} array is updated.
+    \item The \source{} array may be safely reused.
+    \end{itemize}
+
+    Error checking will be done to ensure a valid team handle is provided.
+    All errors are considered fatal and will result in the job aborting
+    with an informative error message.
```

Comment:
There is an open issue #25 to fix this requirement in a number of team operations. It was argued that checking for valid handles will introduce a performance penalty. So the only checks should be for known handle values like SHMEM_TEAM_NULL, etc.

Reply:
Sounds good. Leave this for now, commented or delete it?

Reply:
I have got this description in the team broadcast API call. I haven't gotten feedback yet but it was similar to what got discussed at the last WG call.

    Error checking will be done to detect a value of \LibConstRef{SHMEM\_TEAM\_NULL} passed
    for the team argument. In that case, the program will abort with an informative
    error message. If an invalid team handle is passed to the routine,
    the behavior is undefined.